### PR TITLE
Remove macos-12 from GA kitchen-tests

### PR DESCRIPTION
**Pull request by tpowell-progress**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: kitchen-tests/kitchen.exec.macos.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ lifecycle:
 
 platforms:
   - name: macos-latest # arm64
```
Comment (Stromweld) [inline kitchen-tests/kitchen.exec.macos.yml:29]: for clarity and future proofing when they take 15 out of beta and make it the latest.
```suggestion
  - name: macos-14 # arm64
```

**Review by Stromweld — APPROVED** [review]
not a blocker but a suggestion.

**Comment by sean-simmons-progress** [comment]
@tpowell-progress  can you review the changes suggested by @Stromweld

**Comment by Stromweld** [comment]
I'm sorry I made the suggestion to the wrong file It was supposed to be in the .github/workflows/kitchen.yml file and not the kitchen-tests/kitchen.exec.macos.yml

**Review by Stromweld — APPROVED** [review]
LGTM

**Pull request merged**